diff --git a/web/src/main/resources/db/migration/V0_4__AdminPassword.sql b/web/src/main/resources/db/migration/V0_4__AdminPassword.sql
new file mode 100644
index 0000000..f01f2ad
--- /dev/null
+++ b/web/src/main/resources/db/migration/V0_4__AdminPassword.sql
@@ -0,0 +1 @@
+update users set password = '4e8ec31219936d4b60f793af1923291910778c30179cb954db2eb4a9c8bc34da7516e69f0b422065' where username = 'admin';
\ No newline at end of file
4cc087a1b4731d1017844cc86323df43068b0409 Introduced database schema and data versioning, as well as tidied up test implementations so they can be run repeatedly without deleting the database.
diff --git a/web/src/main/resources/db/migration/V0_2__SeedData.sql b/web/src/main/resources/db/migration/V0_2__SeedData.sql
new file mode 100644
index 0000000..1896efe
--- /dev/null
+++ b/web/src/main/resources/db/migration/V0_2__SeedData.sql
@@ -0,0 +1,10 @@
+
+insert into users (username, password, enabled, email, fullname) values ('admin', 'f8f0cf7ba6486ae73858572eeff1830e67978d0426e7b894ada42e8d0af38a7658a1d1a8ce81ff2e', true, 'admin@mail.com', 'Administrator');
+insert into authorities (username, authority) values ('admin', 'ADMIN');
+
+insert into shiftType (code, description) values ('B1H', 'Early');
+insert into shiftType (code, description) values ('BFH', 'Early');
+insert into shiftType (code, description) values ('C1H', 'Mid');
+insert into shiftType (code, description) values ('CFH', 'Mid');
+insert into shiftType (code, description) values ('S1H', 'Late');
+insert into shiftType (code, description) values ('T1H', 'Late');
